miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/ads/AdMobDisplayer.kt

```diff
@@ -40,81 +55,236 @@ class AdMobDisplayer(private val context: Activity) : MiniAppAdDisplayer, Corout
-                    override fun onAdLoaded() {
-                        onLoaded.invoke()
-                    }
+    private fun loadOldInterstitialAd(
+        adUnitId: String,
+        onLoaded: () -> Unit,
+        onFailed: (String) -> Unit
+    ) {
+        if (interstitialAdMap.containsKey(adUnitId))
+            onFailed.invoke(createLoadReqError(adUnitId))
+        else {
+            val ad = InterstitialAd(context)
+            ad.adUnitId = adUnitId
+            interstitialAdMap[adUnitId] = ad
 
-                    override fun onAdFailedToLoad(adError: LoadAdError) {
-                        interstitialAdMap.remove(adUnitId)
-                        onFailed.invoke(adError.message)
-                    }
+            ad.adListener = object : AdListener() {
+                override fun onAdLoaded() {
+                    onLoaded.invoke()
                 }
 
-                ad.loadAd(AdRequest.Builder().build())
+                override fun onAdFailedToLoad(adError: LoadAdError) {
+                    interstitialAdMap.remove(adUnitId)
+                    onFailed.invoke(adError.message)
+                }
             }
+
+            ad.loadAd(AdRequest.Builder().build())
         }
     }
 
+    /** Load the interstitial ad or version 20.2.0 when it is ready. **/
+    private fun loadLatestInterstitialAd(
+        adUnitId: String,
+        onLoaded: () -> Unit,
+        onFailed: (String) -> Unit
+    ) {
+        AdmobDisplayerLatest.getInstance()
+            .loadInterstitialAd(adUnitId = adUnitId) { loadStatus, errorMessage ->
+                when (loadStatus) {
+                    AdStatus.LOADED -> onLoaded.invoke()
+                    AdStatus.FAILED -> {
+                        interstitialAdMap.remove(adUnitId)
+                        onFailed.invoke(errorMessage ?: "")
+                    }
+                    else -> Log.d("Ad", "Something went wrong.")
+                }
+            }
+    }
+
     /** Show the interstitial ad when it is already loaded. **/
-    override fun showInterstitialAd(adUnitId: String, onClosed: () -> Unit, onFailed: (String) -> Unit) {
+    override fun showInterstitialAd(
+        adUnitId: String,
+        onClosed: () -> Unit,
+        onFailed: (String) -> Unit
+    ) {
         launch {
-            if (interstitialAdMap.containsKey(adUnitId) && interstitialAdMap[adUnitId]!!.isLoaded) {
-                val ad = interstitialAdMap[adUnitId]!!
+            whenHasAdmobLatest { available ->
+                if (available)
+                    showLatestInterstitialAd(adUnitId, onClosed, onFailed)
+                else
+                    showOldInterstitialAd(adUnitId, onClosed, onFailed)
+            }
+        }
+    }
+
+    private fun showOldInterstitialAd(
+        adUnitId: String,
+        onClosed: () -> Unit,
+        onFailed: (String) -> Unit
+    ) {
+        if (interstitialAdMap.containsKey(adUnitId) && interstitialAdMap[adUnitId]!!.isLoaded) {
```

Comment:
You still have a forced unwrapping there ;)

Reply:
updated it,  thanks 🙂